test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerTest.cs

```diff
@@ -274,6 +276,48 @@ public void TempData_CanSetAndGetValues()
             Assert.Equal(input, result);
         }
 
+        public static TheoryData<object, Type> IncompatibleModelData
+        {
+            get
+            {
+                // Small grab bag of instances and expected types with no common base except typeof(object).
+                return new TheoryData<object, Type>
+                {
+                    { null, typeof(object) },
+                    { true, typeof(bool) },
+                    { 43.78, typeof(double) },
+                    { "test string", typeof(string) },
+                    { new List<int>(), typeof(List<int>) },
+                    { new List<string>(), typeof(List<string>) },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IncompatibleModelData))]
+        public void ViewDataModelSetter_DoesNotThrow(object model, Type expectedType)
+        {
+            // Arrange
+            var activator = new ViewDataDictionaryControllerPropertyActivator(new EmptyModelMetadataProvider());
```

Comment:
What would the two tests you're imagining do?

This test ensures the `ViewDataDictionary` exposed on `Controller` has no type restrictions i.e. the declared type is `typeof(object)`. Fits here (a) because we don't have a `ViewDataDictionaryControllerPropertyActivatorTest` class, (b) such a test class would need a `TestableController` and so on, and (c) it's about the `ViewData` property on `Controller`.